extra/import_data.py

```diff
@@ -1,197 +1,158 @@
+    mongoclient = pymongo.MongoClient("mongodb://localhost:27017/")
+    db = mongoclient.vus.lastUpdatedOn
+    
+    # Retrieve the last time MongoDB has updated itself, to filter the latest Alissa analyses only
+    lastUpdatedOn_mongoDB = db.find_one({"version" : 1}, {"lastUpdatedOn":1, "_id":0})
+    #If not exist yet, use '2018-01-01T00:00:00.000+0000' and add to mongo (first time use)
+    if lastUpdatedOn_mongoDB is None:
+        db.replace_one({"version" : 1}, {"lastUpdatedOn" : '2018-01-01T00:00:00.000+0000', "version" : 1}, True)
+        lastUpdatedOn_mongoDB = db.find_one({"version" : 1}, {"lastUpdatedOn":1, "_id":0})
+    
+    for key, value in lastUpdatedOn_mongoDB.items():
+        lastUpdatedOn_mongoDB = value
+    
+    # retrieving data from Alissa
+    analysis = None
+    for analysis in client.get_analyses(status='COMPLETED', 
+                                        lastUpdatedAfter=lastUpdatedOn_mongoDB, # format is '2020-01-01T00:00:00.000+0000'
+                                        analysisPipelineName='ONB01',
+                                        analysisType='INHERITANCE'): #add for testing [:x], where x is numer of iterations
+        if not analysis['classificationTreeName']:  # Skip analysis
+            continue
+        else:
+            # retrieve basic info from Alissa about the analysis
+            patient_dn_no = client.get_report(analysis['id'])[0]['reportName'].split("_")[0]
+            logger.info('start Alissa retrieval of: %s' % patient_dn_no)
+            ONB01_analysis = client.get_inheritance_analyses(analysis['id'])
+            accession_number = client.get_patient(analysis['patientId'])['accessionNumber']
+            export_id = client.post_inheritance_analyses_variants_export(analysis['id'], marked_review=True, marked_include_report=False)['exportId']
+            analyis_sources = client.get_sources(analysis['id'])
+            
+            # retrieve the VUS/GUS info based on the analysis ID
+            VUS_response = None
+            while VUS_response is None:
+                try:
+                    time.sleep(5)  # 5 sec delay to request exported report.
+                    VUS_response = client.get_inheritance_analyses_variants_export(analysis['id'], export_id)
+                    #response.raise_for_status()
+                except HTTPError as exception:
+                    print(exception)
+                    pass
+            logger.info('Alissa retrieval completed of: %s' % patient_dn_no)
+            # sometimes the there are no VUS marked or found within an analysis, then no info needs to be uploaded
+            if VUS_response == []:
+                logger.info('Patient %s, has no VUS marked/found. Not uploaded to MongoDB' % patient_dn_no)
+                continue
+            else:
+                upload_to_mongodb(ONB01_analysis, accession_number, analyis_sources, patient_dn_no, VUS_response)
```

Comment:
If possible I like the 'if' to perform what I want and create a 'positive' test. So I would turn this around and do this:


```python
if vus_response:
    upload....
else:
    log
```

Reply:
I think I tried that actually, but if I remember correctly that this empty list was still considered a response. might want to test this though.

Reply:
I left it like it was, feels more natural like this